src/core/Akka.Persistence/RecoveryPermitter.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using Akka.Actor;
+using Akka.Event;
+using Akka.Pattern;
+using Akka.Persistence.Internal;
+
+namespace Akka.Persistence
+{
+    internal class RequestRecoveryPermit
+    { }
+
+    internal class RecoveryPermitGranted
+    { }
+
+    internal class ReturnRecoveryPermit
+    { }
+
+    /// <summary>
+    /// When starting many persistent actors at the same time the journal its data store is protected 
+    /// from being overloaded by limiting number of recoveries that can be in progress at the same time.
+    /// </summary>
+    internal class RecoveryPermitter : UntypedActor
+    {
+        private readonly int maxPermits;
```

Comment:
Conventions... We will prepare an article about porting a code fron scala in the future